spec/fixtures/sample.js

```diff
@@ -10,4 +10,20 @@ var quicksort = function () {
   };
 
   return sort(Array.apply(this, arguments));
-};
\ No newline at end of file
+};
+
+foo({
+    atIndentTwo: true
```

Comment:
I agree with the desirability of all of the above. I'm just pointing out the trade-off you guys will need to make until we can do all of the above right simultaneously. With the current logic we can't, so we need to pick the lesser evil. Sounds like I may just be an outlier here, although I haven't really seen too many issues regarding too much indentation. Most people seem to complain about inconsistent indentation -- such as indentation that is different before and after a block. That kind of consistency you won't get unless you buy into the counting logic I propose. For me it's a complete deal-breaker if I cannot auto-indent the whole file without risking screwing it up entirely. With all the files I've tested my logic has never done that, but the current one did on virtually any file I tried (the last line always had indentation much greater than zero in the end, which it shouldn't).

Please let me know what you want to do. There is nothing more I can so at this point (other than fix the test case).